main.py

```diff
@@ -15,54 +15,125 @@
-	
-	optimizer = optim.Adam(model.parameters(),lr=hparams['LEARNING_RATE'])
-	criterion = nn.CrossEntropyLoss(ignore_index=dataset.vocab.word_to_index['<PAD>'])
-
-	train(
-		num_epochs=hparams['NUM_EPOCHS'],
-		model=model,
-		train_loader=train_loader,
-		test_loader=test_loader,
-		optimizer= optimizer,
-		criterion=criterion,
-		device=hparams['device'],
-		log_interval=hparams['LOG_INTERVAL']
-	)
+    # Choosing image transformations
+    transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Resize(hparams['image_size']),
+        transforms.CenterCrop(hparams['image_size']),
+        # The normalize parameters depends on the model we're gonna use
+        # If we apply transfer learning from a model that used ImageNet, then
+        # we should use the ImageNet values to normalize the dataset
+        # Otherwise we could just normalize the values between -1 and 1 using the
+        # standard mean and standard deviation
+        # transforms.Normalize(mean=[0.485, 0.456, 0.406],std=[0.229, 0.224, 0.225]),
+        transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])
+    ])
+    # Initializing & splitting dataset
+    dataset = Flickr8kDataset(
+        dataset_folder=CONFIGURATION['DATASET_FOLDER'],
+        transform=transform,
+        reduce=hparams['REDUCE_VOCAB'],
+        vocab_max_size=hparams['VOCAB_SIZE']
+    ).to(device)
+
+    train_split, test_split = random_split(
+        dataset, [32364, 8091])  # 80% train, 20% test
+
+    # Initializing DataLoaders
+    train_loader = DataLoader(
+        train_split,
+        batch_size=hparams['BATCH_SIZE'],
+        collate_fn=CapsCollate(
+            pad_idx=dataset.vocab.word_to_index['<PAD>'], batch_first=True)
+    ).to(device)
+    test_loader = DataLoader(
+        test_split,
+        batch_size=hparams['BATCH_SIZE'],
+        collate_fn=CapsCollate(
+            pad_idx=dataset.vocab.word_to_index['<PAD>'], batch_first=True)
+    ).to(device)
+
+    # instantiating model
+    model = ImageCaptioningModel(
+        embed_size=hparams['EMBED_SIZE'],
+        vocab_size=len(dataset.vocab.word_to_index),
+        caption_max_length=hparams['MAX_LENGTH']
+    ).to(device)
+
+    # select optimizer and criterion
+    optimizer = optim.Adam(model.parameters(), lr=hparams['LEARNING_RATE'])
+    criterion = nn.CrossEntropyLoss(
+        ignore_index=dataset.vocab.word_to_index['<PAD>'])
+
+    # LET'S TRAIN!!!
+    train(
+        num_epochs=hparams['NUM_EPOCHS'],
+        model=model,
+        train_loader=train_loader,
+        test_loader=test_loader,
+        optimizer=optimizer,
+        criterion=criterion,
+        device=hparams['device'],
+        log_interval=['log_interval']
+    )
+
 
 if __name__ == "__main__":
-	main()
-	
\ No newline at end of file
+
+    image_size = (128, 128)
```

Comment:
we should delete all this code right? it's not necessary, instead we just call main()